src/pages/Regions/AddRegion.tsx

```diff
@@ -45,32 +45,34 @@ const AddNewRegionFormPage = () => {
     try {
       setLoading(true);
 
-      if (!await checkIfNameExists(values.regionName)) {
-        const newRegion: RegionProfile = {
-          id: 0,
-          regionName: values.regionName,
-          description: values.description,
-          phoneNumber: values.phoneNumber,
-          email: values.email,
-          link: values.link,
-          logo: logo,
-          street: values.street,
-          houseNumber: values.houseNumber,
-          officeNumber: values.officeNumber,
-          postIndex: values.postIndex,
-          city: values.city,
-          isActive: true
-        };
-        await createRegion(newRegion);
-        form.resetFields();
+      await checkIfNameExists(values.regionName).then(async (response) =>{
```

Comment:
Could this logic be moved into separate file(service e.g) and just called here and in `RegionEdit`?
because it bad way to use something in different places
for e.g what would be if you need to change some property name or something else in that function? you will go and change all occurrence in project?

Reply:
Could not because there are different realization of the ".then" method at the "AddRegion" and "RegionEdit" components